core/lib/contract_verifier/src/tests/real.rs

```diff
@@ -505,7 +598,162 @@ async fn using_real_compiler_in_verifier(bytecode_kind: BytecodeMarker, toolchai
     let (_stop_sender, stop_receiver) = watch::channel(false);
     verifier.run(stop_receiver, Some(1)).await.unwrap();
 
-    assert_request_success(&mut storage, request_id, address, &output.bytecode).await;
+    assert_request_success(&mut storage, request_id, address, &output.bytecode, &[]).await;
+}
+
+#[test_casing(2, [false, true])]
+#[tokio::test]
+async fn using_zksolc_partial_match(use_cbor: bool) {
+    let (compiler_resolver, supported_compilers) = real_resolver!();
+
+    let mut req: VerificationIncomingRequest = VerificationIncomingRequest {
+        compiler_versions: supported_compilers
+            .clone()
+            .solc_for_api(BytecodeMarker::EraVm),
+        ..test_request(Address::repeat_byte(1), COUNTER_CONTRACT)
+    };
+    let hash_type = if use_cbor { "ipfs" } else { "keccak256" };
+    // We need to manually construct the input, since `SolSingleFile` doesn't let us specify metadata hash type.
+    // Note: prior to 1.5.7 field was named `bytecodeHash`.
+    req.source_code_data = SourceCodeData::StandardJsonInput(
+        serde_json::json!({
+            "language": "Solidity",
+            "sources": {
+                "Counter.sol": {
+                    "content": COUNTER_CONTRACT,
+                },
+            },"settings": {
+                "outputSelection": {
+                "*": {
+                    "": [
+                    "abi"
+                    ],
+                    "*": [
+                    "abi"
+                    ]
```

Comment:
Nit: Weird indentation here (I'd format `"": ["abi"]` etc.) and below in `"optimizer"`.